modules/mudletclient.go

```diff
@@ -0,0 +1,181 @@
+package modules
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/GoMudEngine/GoMud/internal/events"
+	"github.com/GoMudEngine/GoMud/internal/mudlog"
+	"github.com/GoMudEngine/GoMud/internal/plugins"
+	"github.com/GoMudEngine/GoMud/internal/users"
+)
+
+var (
+	//go:embed mudletclient/*
+	mudletclient_Files embed.FS
+	
+	// Instance accessible from elsewhere in this package
+	mudletClientModule MudletClientModule
+)
+
+func init() {
+	// Create the module instance first
+	mudletClientModule = MudletClientModule{
+		plug: plugins.New(`mudletclient`, `1.0`),
+	}
+
+	// Attach the file system to the plugin
+	if err := mudletClientModule.plug.AttachFileSystem(mudletclient_Files); err != nil {
+		panic(err)
+	}
+
+	// Set default values directly during initialization
+	mudletClientModule.setDefaultValues()
+
+	// Register the onLoad callback to load config values after plugin system is ready
+	mudletClientModule.plug.Callbacks.SetOnLoad(func() {
+		mudletClientModule.RefreshConfig()
+	})
+
+	// Export a function to register client info for Mudlet clients
+	mudletClientModule.plug.ExportFunction("SendClientInfo", mudletClientModule.SendClientInfoExported)
+}
+
+type MudletClientModule struct {
+	plug          *plugins.Plugin
+	uiUrl         string
+	uiVersion     string
+	mapUrl        string
+	mapperUrl     string
+	mapperVersion string
+	// Track which users already have data sent
+	registeredUsers map[int]bool
+}
+
+// Set hardcoded default values for initialization
+func (g *MudletClientModule) setDefaultValues() {
+	g.uiUrl = ""
+	g.uiVersion = "1.0.0"
+	g.mapUrl = "https://github.com/MorquinDevlar/GoMudUI/releases/latest/download/gomud.dat"
+	g.mapperUrl = "https://github.com/MorquinDevlar/GoMudUI/releases/latest/download/mudlet-mapper.mpackage"
+	g.mapperVersion = "1.0.0"
+	g.registeredUsers = make(map[int]bool)
+}
```

Comment:
These should all be defined in the `moduletclient/data-overlays/config.yaml` - This definition is unnecessary if defaults are defined there. Then they would always be the default values unless someone changed them in the main config, or via `server set`

You can also change module values in your overrides file... As an example, it ends up looking like this in your overrides config file (with your module/values set):

```
Server:
  MudName: GoMud Dev
  NextRoomId: 1003

... etc ...

Modules:
  auctions:
    Anonymous: false
    DurationSeconds: 180
    Enabled: true
    UpdateSeconds: 30
  leaderboards:
    ExperienceEnabled: true
    GoldEnabled: true
    KillsEnabled: true
    Size: 10
```

(registeredUsers obviously wouldn't be a config value.)